class_2.tex

```diff
@@ -106,55 +119,89 @@ \subsection{Lattices and Fixpoints}
+        y \sqsubseteq z$.
+    \end{enumerate}
+    From (ii) we know that $\hat{y}$ is an upper bound, and therefore, by (iii), $\hat{x} \sqsubseteq \hat{y}$. Symmetrically, by (i) and (iv) we get $\hat{y} \sqsubseteq \hat{x}$. By antisymmetry, $x = y$.
+\end{proof}
+
+Given that the \emph{least upper bound} and the \emph{greatest lower bound} are unique (if they exist), for any set $A$ and $B \subseteq A$, we write $\bigsqcup B$ to denote $\lub(B)$ and $\bigsqcap B$ to denote $\glb(B)$.
+
+\begin{definition}[Lattice]
+    A poset $(A,\sqsubseteq)$ is a \emph{lattice} if for all $x$, $y$ in $A$, the lub $\bigsqcup \{x,y\}$ ($x \sqcup y$, join) and the glb $\bigsqcap \{x,y\}$ ($x \sqcap y$, meet) exist.
+\end{definition}
 
 \begin{definition}[Complete Lattice]
-    Poset $(A,\sqsubseteq)$ is a \emph{complete-lattice} if for all $B \subseteq A$, both $\bigsqcap B$ and $\bigsqcup B$ exist.
+    A poset $(A,\sqsubseteq)$ is a \emph{complete lattice} if for all $B \subseteq A$, the lub $\bigsqcup B$ and glb $\bigsqcap B$ exist.
 \end{definition}
+
 \begin{example}
-    Let $(A,\sqsubseteq)$ be a complete-lattice. Then: $\bigsqcup A = \top$, $\bigsqcap A = \bot$, $\bigsqcup \varnothing = \bot$, and $\bigsqcap \varnothing = \top$
+    Let $(A,\sqsubseteq)$ be a complete-lattice. Then: $\bigsqcup A = \top$, $\bigsqcap A = \bot$, $\bigsqcup \varnothing = \bot$, and $\bigsqcap \varnothing = \top$. $\top$ and $\bot$ are called top and bottom, respectively.
 \end{example}
 
-% \begin{definition}
-%     Let $(A,\sqsubseteq)$ be a poset and let $F$ be a function from $A$ to $A$. 
-%     Then $x$ in $A$ is a pre-fixpoint of $F$ if $x \sqsubseteq F(x)$ and is a post-fixpoint of $F$, if $F(x) \sqsubseteq x$. 
-% \end{definition}
-
+\begin{definition}[Monotone Function]
+Let $(A, \sqsubseteq)$ be a poset. A function $F: A \to A$ is \emph{monotone} if for all $x$, $y$ in $A$, if $x \sqsubseteq y$ then $F(x) \sqsubseteq F(y)$.
+\end{definition}
 
+\begin{definition}
+Let $(A, \sqsubseteq)$ be a poset and $F: A \to A$. An element $x$ in $A$ is,
+\begin{enumerate}
+  \item a \emph{prefixpoint} of $F$ if $x \sqsubseteq F(x)$,
+  \item a \emph{postfixpoint} of $F$ if $F(x) \sqsubseteq x$.
+\end{enumerate}
+\end{definition}
 
+% TODO
+\begin{definition}[Fixpoint]
+Let $(A, \sqsubseteq)$ be a poset and $F: A \to A$. An element $x$ in $A$ is a \emph{fixpoint} of $F$ if it is both a \emph{pre-} and \emph{postfixpoint} of $F$.
+\end{definition}
 
 \begin{theorem}[Knaster-Tarski]
 \label{thrm:knaster}
-    For every complete lattice $(A,\sqsubseteq)$ and monotone function $f$ on $A$, it holds that: (i) $\bigsqcup \{x\in A| x \sqsubseteq f(x)\}$ is the unique greatest fixpoint of $F$; (ii) $\bigsqcap \{x \in A| f(x) \sqsubseteq x\}$ is the unique least fixpoint of $f$. 
+    Let $(A,\sqsubseteq)$ be a complete lattice and let $F$ a monotone function on $A$.
+    Further, let $\hat{y} = \{x \mid x \sqsubseteq F(x)\}$ and $\hat{z} = \{x \mid F(x) \sqsubseteq x\}$.
```

Comment:
here you are saying that \hat{y} and \hat{z} are subsets of A but below you are saying they are fixpoints